src/app.cpp

```diff
@@ -36,23 +36,14 @@ App::~App() { MZ_COUNT_DTOR(App); }
 
 int App::state() const { return m_state; }
 
-App::UserState App::userState() const { return m_userState; }
-
 void App::setState(int state) {
   logger.debug() << "Set state:" << state;
 
   m_state = state;
+  emit userAuthenticationChanged();
```

Comment:
Why is this emitted here? Isn't it possible the user state didn't change?

Reply:
There is no longer a `userState` to change, but the `userAuthenticated` property is derived from the state now, so when the state changes we need to emit a notification that `userAuthenticated` may have changed value as well.

Reply:
Agreed fully, though I missed that we check if user is authenticated in the one place we listen to that signal.

I think we should rename this to `userAuthenticationMaybeChanged` or something like that - to make clear that's its not guaranteed that the state changed when that is emitted.

Reply:
Done!